tests/plugins/test_quotes.py

```diff
@@ -33,10 +33,32 @@
     ),
 )
 def test_to_html_quotebox_chat(testquote: str, n_parsed_lines: int) -> None:
+    """Tests that chatlines are splitted correctly (or left unparsed)."""
     quotebox = quotes.to_html_quotebox(testquote)
     assert quotebox.count('<div class="chat">') == n_parsed_lines
 
 
+@pytest.mark.parametrize(
+    "testquote, is_colorized",
+    (
+        # Parsed, single-line quotes.
+        # Unparsed quotes would produce consistent results but are already tested by
+        # `test_to_html_quotebox_chat`.
+        ("[21:33] reg: testline", True),
+        ("[21:33] @auth: testline", True),
+        ("[21:33] reg: hey: +how: are: @you", True),
+        ("reg: auth", False),
+        ("[21:33] (Plato notes: reason)", False),
+    ),
+)
+def test_to_html_quotebox_colorize(testquote: str, is_colorized: bool) -> None:
+    """Tests that usernames are colorized only once per line and if necessary."""
+    quotebox = quotes.to_html_quotebox(testquote)
+    tag_count = int(is_colorized)  # Occurrences of <username>, at most 1.
```

Comment:
This line may seem a bit obscure I guess, but using the boolean `is_colorized` for a logic condition seems overall more readable than using a count integer!